### PR TITLE
HEXDEV-700: Hadoop driver chooses deterministically the node to connect to

### PR DESCRIPTION
**Pull request by michalkurka**
This change makes sure that the node users will connect to on Hadoop is chosen deterministically.

This change is only targeted to fix release. The stable release might introduce a breaking change in the API.

@mmalohlava Will need a change in SW as well.

### CODE REVIEW

File: h2o-core/src/main/java/water/init/AbstractEmbeddedH2OConfig.java [hunk 1]
```diff
@@ -38,16 +38,31 @@
...
    * a cloud of a certain size.
    * This may be nonblocking.
    *
+   * Note: this method will be made abstract in a future stable release. User is expected to override the method
+   * in the implementation.
+   *
    * @param ip IP address this H2O can be reached at.
    * @param port Port this H2O can be reached at (for REST API and browser).
+   * @param leaderIp IP address of the leader H2O node of the cloud.
+   * @param leaderPort Port of the leader H2O node (for REST API and browser).
```
Comment (mmalohlava) [inline h2o-core/src/main/java/water/init/AbstractEmbeddedH2OConfig.java]: What if there is no REST API exposed?

**Comment by mmalohlava** [comment]
CC: @jakubhava

File: h2o-core/src/main/java/water/init/AbstractEmbeddedH2OConfig.java (hunk 1 above)
Comment (michalkurka) [reply h2o-core/src/main/java/water/init/AbstractEmbeddedH2OConfig.java]: Hmm, a case when the leader doesn't have REST API exposed but the current member does?

Such clusters cannot be created from the  hadoop driver but they are possible in general. Do we need to consider this case?

Comment (mmalohlava) [reply h2o-core/src/main/java/water/init/AbstractEmbeddedH2OConfig.java]: That's case of SW - we are using EmbeddedH2OConfig, but running each executor h2o with `--disable_web`.

Comment (michalkurka) [reply h2o-core/src/main/java/water/init/AbstractEmbeddedH2OConfig.java]: But that is still okay (or at least consistent with current behavior).

Currently we report embeddedWebServerIp/Port regardless if the node has enabled REST or it doesn't.

Comment (mmalohlava) [reply h2o-core/src/main/java/water/init/AbstractEmbeddedH2OConfig.java]: oki, so no harm if we send bogus port

Comment (michalkurka) [reply h2o-core/src/main/java/water/init/AbstractEmbeddedH2OConfig.java]: Yes, I think that is the case.

There will only be an issue if the cluster was not homogeneous (some non-client nodes started with `--disable_web` and some not). Although this is theoretically possible I don't think we need to be concerned with the use case.

**Pull request merged**